### PR TITLE
Add biconnectivity module

### PR DESCRIPTION
**Pull request by abhinavmehndiratta**
Implemented iterative `biconnected_components` to avoid StackOverflow. No significant change in performance -
New -
```julia
julia> g = loadsnap(:facebook_combined)
{4039, 88234} undirected simple Int64 graph

julia> @btime LightGraphs.Biconnectivity.biconnected_components(g)
  3.686 ms (232 allocations: 4.55 MiB)
```
Old -
```julia
julia> g = loadsnap(:facebook_combined)
{4039, 88234} undirected simple Int64 graph

julia> @btime biconnected_components(g)
  3.779 ms (231 allocations: 4.46 MiB)
```

### CODE REVIEW

**Comment by abhinavmehndiratta** [comment]
We should also update doc examples since many functions like `star_graph` are deprecated.

**Comment by abhinavmehndiratta** [comment]
Build failure is unrelated.

File: src/Biconnectivity/articulation.jl [hunk 1]
```diff
@@ -79,8 +79,10 @@ function articulation end
 
     articulation_points = Vector{T}()
 
-    for u in findall(is_articulation_pt)
-        push!(articulation_points, T(u))
+    for u in vertices(g)
+        if is_articulation_pt[u]
+            push!(articulation_points, u)
+        end
```
Comment (sbromberger) [inline src/Biconnectivity/articulation.jl:85]: Is there a way to do this in one statement, perhaps using `filter`?

File: src/Biconnectivity/biconnect.jl [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    nvg = nv(g)
+    timer = one(T)
+    disc = zeros(T, nvg)
+    low = Vector{T}(undef, nvg)
+    stk = Vector{Tuple{T, T, T}}()
+    sizehint!(stk, nvg)
+    track = Vector{SimpleEdge{T}}()
+    bcc = Vector{Vector{SimpleEdge{T}}}()
+
+    for w in vertices(g)
```
Comment (sbromberger) [inline src/Biconnectivity/biconnect.jl:38]: `@inbounds`

File: src/Biconnectivity/articulation.jl (hunk 1 above)
Comment (abhinavmehndiratta) [reply src/Biconnectivity/articulation.jl:85]: I don't think `filter` or `findall` support converting to a type, we could do a `findall` followed by a `convert`, but that would allocate more memory.

Comment (sbromberger) [reply src/Biconnectivity/articulation.jl:85]: where are we type-converting? Also, just noticed that `is_articulation_pt` is a bitvec, so...
```
articulation_points = vertices(g)[is_articulation_point]
```
should do it?

**Pull request merged**